cdap-ui/app/cdap/components/FieldLevelLineage/v2/Context/FllContext.tsx

```diff
@@ -17,46 +17,131 @@
 import React from 'react';
 import data from './sample_response';
 import {
-  parseRelations,
-  makeTargetNodes,
+  makeTargetFields,
+  IField,
+  ILink,
+  ITableFields,
+  getFieldsAndLinks,
 } from 'components/FieldLevelLineage/v2/Context/FllContextHelper';
+import * as d3 from 'd3';
 
-export interface INode {
-  id: string;
-  name: string;
-  group: number;
+export const FllContext = React.createContext<IContextState>({} as IContextState);
+
+export interface IContextState {
+  target: string;
+  targetFields: IField[];
+  links: ILink[];
+  causeSets: ITableFields;
+  impactSets: ITableFields;
+  activeField: string;
+  activeCauseSets: ITableFields;
+  activeImpactSets: ITableFields;
+  activeLinks: ILink[];
+  showingOneField: boolean;
+  numTables: number;
+  firstCause: number;
+  firstImpact: number;
+  firstField: number;
+  handleFieldClick: (event: React.MouseEvent<HTMLDivElement>) => void;
+  handleViewCauseImpact: () => void;
+  handleReset: () => void;
 }
 
-const FllContext = React.createContext({});
+export class Provider extends React.Component<{ children }, IContextState> {
+  private parsedRes = getFieldsAndLinks(data);
 
-function getFieldsAndLinks(d) {
-  const incoming = parseRelations(d.entityId.namespace, d.entityId.dataset, d.incoming);
-  const outgoing = parseRelations(d.entityId.namespace, d.entityId.dataset, d.outgoing, false);
-  const causeTables = incoming.tables;
-  const impactTables = outgoing.tables;
-  const nodes = incoming.relNodes.concat(outgoing.relNodes);
-  const links = incoming.relLinks.concat(outgoing.relLinks);
-  return { causeTables, impactTables, nodes, links };
-}
+  private handleFieldClick = (e) => {
+    const activeField = e.target.id;
+    if (!activeField) {
+      return;
+    }
+    d3.select(`#${this.state.activeField}`).classed('selected', false);
+
+    this.setState(
+      {
+        activeField,
+      },
+      () => {
+        d3.select(`#${activeField}`).classed('selected', true);
+        this.getActiveLinks();
+      }
+    );
+  };
+
+  private getActiveLinks() {
+    const activeFieldId = this.state.activeField;
+    const activeLinks = [];
+    this.state.links.forEach((link) => {
+      const isSelected = link.source.id === activeFieldId || link.destination.id === activeFieldId;
+      if (isSelected) {
+        activeLinks.push(link);
+      }
+    });
+    this.setState({ activeLinks });
+  }
+
+  private getActiveSets() {
```

Comment:
I am not entirely sure what the function is doing but wouldn't `activeCauseSets` & `activeImpactSets` always be `{}` when we do `setState`?

Reply:
This function is looking at the active links and filtering all the cause and impact sets to only include the sets and fields that are "active" (for when someone clicks on "View cause and impact sets"). activeCauseSets and activeImpactSets might not be empty objects if they clicked on "view cause and impact sets" previously.

Reply:
Will update in future PR, as we discussed!